The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9998078923/job/27636173430) [(plain)](https://github.com/rust-lang/rust/commit/cc4b217d13fd6ba1e3e678724706380548ee5e41/checks/27636173430/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:3aacb9c90579defe09351ac5e8ee504359f8054da6326ff19038f1b7c90e3cb2aafe33685c6d9b76ee8d2ccbd187ca80c46ab5380485abdd8c0ce7d69cd8d8fd:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---

---- [ui] tests/ui/asm/binary_asm_labels.rs stdout ----
diff of stderr:

4 LL |         asm!("0: jmp 0b");
5    |               ^ use a different label that doesn't start with `0` or `1`
6    |
-    = note: an LLVM bug makes these labels ambiguous with a binary literal number
+    = help: Start numbering with `2` instead
+    = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
8    = note: `#[deny(binary_asm_labels)]` on by default
10 error: avoid using labels containing only the digits `0` and `1` in inline assembly


13 LL |         asm!("1: jmp 1b");
14    |               ^ use a different label that doesn't start with `0` or `1`
15    |
-    = note: an LLVM bug makes these labels ambiguous with a binary literal number
+    = help: Start numbering with `2` instead
+    = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
18 error: avoid using labels containing only the digits `0` and `1` in inline assembly
19   --> $DIR/binary_asm_labels.rs:13:15


21 LL |         asm!("10: jmp 10b");
22    |               ^^ use a different label that doesn't start with `0` or `1`
23    |
-    = note: an LLVM bug makes these labels ambiguous with a binary literal number
+    = help: Start numbering with `2` instead
+    = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
26 error: avoid using labels containing only the digits `0` and `1` in inline assembly
27   --> $DIR/binary_asm_labels.rs:14:15


29 LL |         asm!("01: jmp 01b");
30    |               ^^ use a different label that doesn't start with `0` or `1`
31    |
-    = note: an LLVM bug makes these labels ambiguous with a binary literal number
+    = help: Start numbering with `2` instead
+    = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
34 error: avoid using labels containing only the digits `0` and `1` in inline assembly
35   --> $DIR/binary_asm_labels.rs:15:15


37 LL |         asm!("1001101: jmp 1001101b");
38    |               ^^^^^^^ use a different label that doesn't start with `0` or `1`
39    |
-    = note: an LLVM bug makes these labels ambiguous with a binary literal number
+    = help: Start numbering with `2` instead
+    = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
42 error: aborting due to 5 previous errors
43 



The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/binary_asm_labels/binary_asm_labels.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args asm/binary_asm_labels.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/asm/binary_asm_labels.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/binary_asm_labels" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/binary_asm_labels/auxiliary"
--- stderr -------------------------------
error: avoid using labels containing only the digits `0` and `1` in inline assembly
##[error]  --> /checkout/tests/ui/asm/binary_asm_labels.rs:11:15
   |
   |
LL |         asm!("0: jmp 0b"); //~ ERROR avoid using labels containing only the digits
   |               ^ use a different label that doesn't start with `0` or `1`
   = help: Start numbering with `2` instead
   = help: Start numbering with `2` instead
   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
   = note: `#[deny(binary_asm_labels)]` on by default
error: avoid using labels containing only the digits `0` and `1` in inline assembly
##[error]  --> /checkout/tests/ui/asm/binary_asm_labels.rs:12:15
   |
   |
LL |         asm!("1: jmp 1b"); //~ ERROR avoid using labels containing only the digits
   |               ^ use a different label that doesn't start with `0` or `1`
   = help: Start numbering with `2` instead
   = help: Start numbering with `2` instead
   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
error: avoid using labels containing only the digits `0` and `1` in inline assembly
##[error]  --> /checkout/tests/ui/asm/binary_asm_labels.rs:13:15
   |
   |
LL |         asm!("10: jmp 10b"); //~ ERROR avoid using labels containing only the digits
   |               ^^ use a different label that doesn't start with `0` or `1`
   = help: Start numbering with `2` instead
   = help: Start numbering with `2` instead
   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
error: avoid using labels containing only the digits `0` and `1` in inline assembly
##[error]  --> /checkout/tests/ui/asm/binary_asm_labels.rs:14:15
   |
   |
LL |         asm!("01: jmp 01b"); //~ ERROR avoid using labels containing only the digits
   |               ^^ use a different label that doesn't start with `0` or `1`
   = help: Start numbering with `2` instead
   = help: Start numbering with `2` instead
   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
error: avoid using labels containing only the digits `0` and `1` in inline assembly
##[error]  --> /checkout/tests/ui/asm/binary_asm_labels.rs:15:15
   |
   |
LL |         asm!("1001101: jmp 1001101b"); //~ ERROR avoid using labels containing only the digits
   |               ^^^^^^^ use a different label that doesn't start with `0` or `1`
   = help: Start numbering with `2` instead
   = help: Start numbering with `2` instead
   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
error: aborting due to 5 previous errors
------------------------------------------


```

</details>

--- a/compiler/rustc_lint/messages.ftl
+++ b/compiler/rustc_lint/messages.ftl
@@ -403,8 +403,9 @@ lint_inner_macro_attribute_unstable = inner macro attributes are unstable
 
 lint_invalid_asm_label_binary = avoid using labels containing only the digits `0` and `1` in inline assembly
     .label = use a different label that doesn't start with `0` or `1`
-    .note = an LLVM bug makes these labels ambiguous with a binary literal number
-    .note = see <https://bugs.llvm.org/show_bug.cgi?id=36144> for more information
+    .help = start numbering with `2` instead
+    .note1 = an LLVM bug makes these labels ambiguous with a binary literal number on x86
+    .note2 = see <https://github.com/llvm/llvm-project/issues/99547> for more information
 
 lint_invalid_asm_label_format_arg = avoid using named labels in inline assembly
     .help = only local labels of the form `<number>:` should be used in inline asm

--- a/compiler/rustc_lint/src/builtin.rs
+++ b/compiler/rustc_lint/src/builtin.rs
@@ -2740,8 +2740,9 @@ declare_lint! {
     ///
     /// ### Example
     ///
-    /// ```rust,compile_fail
-    /// # #![feature(asm_experimental_arch)]
+    /// ```rust,ignore (fails on non-x86_64)
+    /// #![cfg(target_arch = "x86_64")]
+    ///
     /// use std::arch::asm;
     ///
     /// fn main() {
@@ -2751,19 +2752,32 @@ declare_lint! {
     /// }
     /// ```
     ///
-    /// {{produces}}
+    /// This will produce:
+    ///
+    /// ```text
+    /// error: avoid using labels containing only the digits `0` and `1` in inline assembly
+    ///  --> <source>:7:15
+    ///   |
+    /// 7 |         asm!("0: jmp 0b");
+    ///   |               ^ use a different label that doesn't start with `0` or `1`
+    ///   |
+    ///   = help: start numbering with `2` instead
+    ///   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+    ///   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
+    ///   = note: `#[deny(binary_asm_labels)]` on by default
+    /// ```
     ///
     /// ### Explanation
     ///
-    /// A [LLVM bug] causes this code to fail to compile because it interprets the `0b` as a binary
-    /// literal instead of a reference to the previous local label `0`. Note that even though the
-    /// bug is marked as fixed, it only fixes a specific usage of intel syntax within standalone
-    /// files, not inline assembly. To work around this bug, don't use labels that could be
-    /// confused with a binary literal.
+    /// An [LLVM bug] causes this code to fail to compile because it interprets the `0b` as a binary
+    /// literal instead of a reference to the previous local label `0`. To work around this bug,
+    /// don't use labels that could be confused with a binary literal.
+    ///
+    /// This behavior is platform-specific to x86 and x86-64.
     ///
     /// See the explanation in [Rust By Example] for more details.
     ///
-    /// [LLVM bug]: https://bugs.llvm.org/show_bug.cgi?id=36144
+    /// [LLVM bug]: https://github.com/llvm/llvm-project/issues/99547
     /// [Rust By Example]: https://doc.rust-lang.org/nightly/rust-by-example/unsafe/asm.html#labels
     pub BINARY_ASM_LABELS,
     Deny,

--- a/compiler/rustc_lint/src/lints.rs
+++ b/compiler/rustc_lint/src/lints.rs
@@ -2066,7 +2066,9 @@ pub enum InvalidAsmLabel {
         missing_precise_span: bool,
     },
     #[diag(lint_invalid_asm_label_binary)]
-    #[note]
+    #[help]
+    #[note(lint_note1)]
+    #[note(lint_note2)]
     Binary {
         #[note(lint_invalid_asm_label_no_span)]
         missing_precise_span: bool,

--- a/tests/ui/asm/binary_asm_labels.stderr
+++ b/tests/ui/asm/binary_asm_labels.stderr
@@ -4,7 +4,9 @@ error: avoid using labels containing only the digits `0` and `1` in inline assem
 LL |         asm!("0: jmp 0b");
    |               ^ use a different label that doesn't start with `0` or `1`
    |
-   = note: an LLVM bug makes these labels ambiguous with a binary literal number
+   = help: start numbering with `2` instead
+   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
    = note: `#[deny(binary_asm_labels)]` on by default
 
 error: avoid using labels containing only the digits `0` and `1` in inline assembly
@@ -13,7 +15,9 @@ error: avoid using labels containing only the digits `0` and `1` in inline assem
 LL |         asm!("1: jmp 1b");
    |               ^ use a different label that doesn't start with `0` or `1`
    |
-   = note: an LLVM bug makes these labels ambiguous with a binary literal number
+   = help: start numbering with `2` instead
+   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
 
 error: avoid using labels containing only the digits `0` and `1` in inline assembly
   --> $DIR/binary_asm_labels.rs:13:15
@@ -21,7 +25,9 @@ error: avoid using labels containing only the digits `0` and `1` in inline assem
 LL |         asm!("10: jmp 10b");
    |               ^^ use a different label that doesn't start with `0` or `1`
    |
-   = note: an LLVM bug makes these labels ambiguous with a binary literal number
+   = help: start numbering with `2` instead
+   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
 
 error: avoid using labels containing only the digits `0` and `1` in inline assembly
   --> $DIR/binary_asm_labels.rs:14:15
@@ -29,7 +35,9 @@ error: avoid using labels containing only the digits `0` and `1` in inline assem
 LL |         asm!("01: jmp 01b");
    |               ^^ use a different label that doesn't start with `0` or `1`
    |
-   = note: an LLVM bug makes these labels ambiguous with a binary literal number
+   = help: start numbering with `2` instead
+   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
 
 error: avoid using labels containing only the digits `0` and `1` in inline assembly
   --> $DIR/binary_asm_labels.rs:15:15
@@ -37,7 +45,9 @@ error: avoid using labels containing only the digits `0` and `1` in inline assem
 LL |         asm!("1001101: jmp 1001101b");
    |               ^^^^^^^ use a different label that doesn't start with `0` or `1`
    |
-   = note: an LLVM bug makes these labels ambiguous with a binary literal number
+   = help: start numbering with `2` instead
+   = note: an LLVM bug makes these labels ambiguous with a binary literal number on x86
+   = note: see <https://github.com/llvm/llvm-project/issues/99547> for more information
 
 error: aborting due to 5 previous errors
 

